lib/ansible/modules/pip.py

```diff
@@ -123,7 +123,7 @@
     check_mode:
         support: full
     diff_mode:
-        support: none
+        support: partial
```

Comment:
@pkingstonxyz you might want to look into the diff mode support once #85623 is in. It's probably worth exploring regardless.